src/main/java/org/cbioportal/legacy/service/impl/GenericAssayServiceImpl.java

```diff
@@ -157,8 +166,18 @@ public List<GenericAssayData> fetchGenericAssayData(List<String> molecularProfil
             Collectors.groupingBy(GenericAssayMolecularAlteration::getMolecularProfileId));
         
         for (Sample sample : samples) {
-            for (MolecularProfile molecularProfile : molecularProfileMapByStudyId.get(sample.getCancerStudyIdentifier())) {
+            List<MolecularProfile> studyMolecularProfiles = molecularProfileMapByStudyId.get(sample.getCancerStudyIdentifier());
+            if (studyMolecularProfiles == null) {
+                throw new IllegalArgumentException("MolecularProfiles for StudyId " + sample.getCancerStudyIdentifier() + " is null.");
+            }
+            for (MolecularProfile molecularProfile : studyMolecularProfiles) {
                 String molecularProfileId = molecularProfile.getStableId();
+                if (internalSampleIdsMap.get(molecularProfileId) == null) {
+                    throw new IllegalArgumentException("InternalSampleIdsMap for MolecularProfileId " + molecularProfileId + " is null.");
+                }
+                if (sample.getInternalId() == null) {
+                    throw new IllegalArgumentException("InternalId for Sample " + sample.getStableId() + " is null.");
```

Comment:
Is this should be `sample. getInternalId()`? Other than that it looks good to me
```suggestion
                    throw new IllegalArgumentException("InternalId for Sample " + sample.getInternalId() + " is null.");
```